### PR TITLE
Wallet & Support for non-credit card payment sources - rebased on solidus:master

### PR DESCRIPTION
**Pull request by peterberkenbosch**
work done in #1091 was on different base repositories and I had some issues getting master rebased in there. This PR is mostly here to see if this is making the CI and Hound happy again.

### CODE REVIEW

File: core/config/locales/en.yml [hunk 1]
```diff
@@ -577,6 +577,10 @@ en:
           attributes:
             user_id:
               default_address_exists: "already has a default address"
+        spree/wallet_payment_source:
+          attributes:
+            payment_source:
+              has_to_be_payment_source_class: "has to be a Spree::PaymentSource"
```
Comment (bbuchalter) [inline core/config/locales/en.yml]: Will this text ever be shown to a user?

File: CHANGELOG.md [hunk 2]
```diff
@@ -87,6 +87,7 @@
     * JavaScript for it has been moved from address.js into its own `spree/frontend/checkout/coupon-code`
     * Numerous small nuisances have been fixed [#1090](https://github.com/solidusio/solidus/pull/1090)
 
+<<<<<<< 356ed069e972d5bae6275b433e17b98f885b3fc4
```
Comment (bbuchalter) [inline CHANGELOG.md]: Not sure we want this line in?

File: core/config/locales/en.yml (hunk 1 above)
Comment (peterberkenbosch) [inline core/config/locales/en.yml]: Not sure yet, mostly here to solve the missing translation error we got in the model when the validation fails.

**Comment by jordan-brough** [comment]
Updating `Gateway#reusable_sources` seems like a good thing to do here as well, as per this PR: https://github.com/solidusio/solidus/pull/1442#issuecomment-248034253

**Comment by peterberkenbosch** [comment]
Close this in favour of #1707. Thanks @chrisradford

**Pull request closed without merging**